workflow/scripts/plot-primer-clipping.py

```diff
@@ -0,0 +1,151 @@
+        for pair in mate_pair_intervals:
+            if (
+                len(mate_pair_intervals[pair]) > 1
+                and mate_pair_intervals[pair][0] != None
+                and mate_pair_intervals[pair][1] != None
+            ):
+                if primer_intervals.envelop(
+                    mate_pair_intervals[pair][0], mate_pair_intervals[pair][1] + 1
+                ):
+                    if (
+                        sorted(
+                            primer_intervals.envelop(
+                                mate_pair_intervals[pair][0],
+                                mate_pair_intervals[pair][1] + 1,
+                            )
+                        )[0].begin
+                        == mate_pair_intervals[pair][0]
+                        and sorted(
+                            primer_intervals.envelop(
+                                mate_pair_intervals[pair][0],
+                                mate_pair_intervals[pair][1] + 1,
+                            )
+                        )[0].end
+                        == mate_pair_intervals[pair][1] + 1
+                    ):
+                        counter_primer += 1
+                    else:
+                        counter_primer_within += 1
+                elif no_primer_intervals.envelop(
+                    mate_pair_intervals[pair][0] + 1, mate_pair_intervals[pair][1]
+                ):
+                    if (
+                        sorted(
+                            no_primer_intervals.envelop(
+                                mate_pair_intervals[pair][0] + 1,
+                                mate_pair_intervals[pair][1],
+                            )
+                        )[0].begin
+                        == mate_pair_intervals[pair][0] + 1
+                        and sorted(
+                            no_primer_intervals.envelop(
+                                mate_pair_intervals[pair][0] + 1,
+                                mate_pair_intervals[pair][1],
+                            )
+                        )[0].end
+                        == mate_pair_intervals[pair][1]
+                    ):
+                        counter_no_primer += 1
+                    else:
+                        counter_no_primer_within += 1
+                else:
+                    counter_nothing += 1
+            else:
+                counter_nothing += 1
+        counters = pd.DataFrame(
+            {
+                "n_count": [
+                    counter_primer,
+                    counter_primer_within,
+                    counter_no_primer,
+                    counter_no_primer_within,
+                    counter_nothing,
+                ],
+                "class": [
+                    "w/ primer",
+                    "w/ primer within",
+                    "w/o primer",
+                    "w/o primer within",
+                    "none",
```

Comment:
What is to be understood by ""w/ primer within"? Maybe we could display more meaningful names :)

Reply:
I wouldn't mess with the names here too much to not get too long names in the plots